packages/main/src/plugin/authentication.ts

```diff
@@ -75,6 +78,64 @@ export interface SessionRequestInfo {
 }
 
 export class AuthenticationImpl {
+  async showAccountsMenu(x: number, y: number): Promise<void> {
+    const template: (MenuItemConstructorOptions | MenuItem)[] = [
+      {
+        label: 'Manage authentication',
+        click: (): void => {
+          this.apiSender.send('navigate', {
+            page: 'authentication-providers',
+          });
+        },
+      },
+      {
+        type: 'separator',
+      } as MenuItem,
+    ];
+
+    this.getSessionRequests().forEach(request => {
+      const provider = this._authenticationProviders.get(request.providerId);
```

Comment:
I  think the whole method will make it hard to move the code to the renderer part because here the backend side is doing all the stuff (UI calls and "backend side")

Could we change to have a separate object/class that will do only UI stuff (I suppose this class should not have any BrowserWindow as it should do only authentication stuff, no menu display)

then, when moving the menu to the renderer part this class/file should be untouched

Reply:
so why not using a separate class that will do this rather than mixing client side and server side in authentication.ts ?

Reply:
I moved out Menu and BrowserWindow related code to authentication-menu.ts
Tests are updates as well.